include/vhost.hrl

```diff
@@ -0,0 +1,30 @@
+-include("vhost_v1.hrl").
+-include("vhost_v2.hrl").
+
+-define(is_vhost(V),
+        (?is_vhost_v2(V) orelse
+         ?is_vhost_v1(V))).
+
+-define(vhost_field_name(Q),
+        case ?is_vhost_v2(Q) of
+            true  -> ?vhost_v2_field_name(Q);
+            false -> case ?is_vhost_v1(Q) of
+                         true -> ?vhost_v1_field_name(Q)
+                     end
+        end).
```

Comment:
In `amqqueue.hrl`, the macros are meant to be used in guard expressions. A `case` statement won't work in this case. Here are examples from that header:

```erlang
-define(amqqueue_is_durable(Q),
        ((?is_amqqueue_v2(Q) andalso
          ?amqqueue_v2_field_durable(Q) =:= true) orelse
         (?is_amqqueue_v1(Q) andalso
          ?amqqueue_v1_field_durable(Q) =:= true))).

-define(amqqueue_exclusive_owner_is(Q, Owner),
        ((?is_amqqueue_v2(Q) andalso
          ?amqqueue_v2_field_exclusive_owner(Q) =:= Owner) orelse
         (?is_amqqueue_v1(Q) andalso
          ?amqqueue_v1_field_exclusive_owner(Q) =:= Owner))).
```

Furthermore, function calls should be used otherwise:
1. That's the point of having a wrapper module, we can change that module and old code calling it still works. If people use macros to access the record (i.e. compile-time syntax like records), they are still screwed: they will have to recompile their code.
2. It's usually better to let the compiler decide what should be inlined or not.

Reply:
It was adopted from `amqqueue.hrl`:

``` erlang
-define(amqqueue_field_name(Q),
        case ?is_amqqueue_v2(Q) of
            true  -> ?amqqueue_v2_field_name(Q);
            false -> case ?is_amqqueue_v1(Q) of
                         true -> ?amqqueue_v1_field_name(Q)
                     end
        end).
```

The above approach returns a boolean, which is not the case with these macros. I have replaced them with functions.